src/collection-preferences/content-display/use-drag-and-drop-reorder.ts

```diff
@@ -0,0 +1,178 @@
+      event.stopPropagation();
+    }
+  };
+
+  const getClosestId = (active: Active) => {
+    if (positionDelta.current === 0) {
+      return active.id;
+    }
+    const currentIndex = sortedOptions.findIndex(({ id }) => id === active.id);
+    const newIndex = Math.max(0, Math.min(sortedOptions.length - 1, currentIndex + positionDelta.current));
+    return sortedOptions[newIndex].id;
+  };
+
+  const collisionDetection: CollisionDetection = ({
+    active,
+    collisionRect,
+    droppableContainers,
+    droppableRects,
+    pointerCoordinates,
+  }) => {
+    if (isKeyboard.current) {
+      const collidingContainerId = getClosestId(active);
+      if (collidingContainerId !== active.id) {
+        const collidingContainer = droppableContainers.find(({ id }) => id === collidingContainerId);
+        if (collidingContainer) {
+          return [
+            {
+              id: collidingContainer.id,
+              data: {
+                droppableContainer: collidingContainer,
+                value: 0,
+              },
+            },
+          ];
+        }
+      }
+      return [];
+    } else {
+      // For mouse usage, the closest center algorithm works fine
+      return closestCenter({ active, collisionRect, droppableRects, droppableContainers, pointerCoordinates });
+    }
+  };
+
+  const coordinateGetter: KeyboardCoordinateGetter = (
+    event,
+    { context: { active, collisionRect, droppableRects, droppableContainers } }
+  ) => {
+    if (event.code === KeyboardCode.Up || event.code === KeyboardCode.Down) {
+      event.preventDefault();
+
+      if (!active || !collisionRect) {
+        return;
+      }
+
+      const closestId = getClosestId(active);
+
+      if (closestId !== null) {
+        const activeDroppable = droppableContainers.get(active.id);
+        const newDroppable = droppableContainers.get(closestId);
+        const newRect = newDroppable ? droppableRects.get(newDroppable.id) : null;
+        const newNode = newDroppable?.node.current;
+
+        if (newNode && newRect && activeDroppable && newDroppable) {
+          const isAfterActive = isAfter(activeDroppable, newDroppable);
+          const offset = {
+            x: isAfterActive ? collisionRect.width - newRect.width : 0,
+            y: isAfterActive ? collisionRect.height - newRect.height : 0,
+          };
+          const rectCoordinates = {
+            x: newRect.left,
+            y: newRect.top,
+          };
+
+          return {
+            x: rectCoordinates.x - offset.x,
+            y: rectCoordinates.y - offset.y,
+          };
+        }
+      }
+    }
+  };
+
+  const sensors = useSensors(
+    useSensor(PointerSensor),
+    useSensor(KeyboardSensor, {
+      coordinateGetter,
+      onActivation: () => {
+        isKeyboard.current = true;
+      },
+    })
+  );
+
+  return {
+    activeItem,
+    collisionDetection,
+    coordinateGetter,
+    handleKeyDown,
+    isKeyboard: isKeyboard.current,
```

Comment:
This is kind of intentional to avoid eventual unnecessary re-renders. When `isKeyboard` changes we don't need tp re-render the entire ContentDisplay component, it is only used by SortableItem, in combination with other state variables that it gets from the `useSortable` hook provided by dnd-kit.

Reply:
Yes, I see how it works now but it might lead to some trouble with future updates. E.g. if someone would use isKeyword in an async handler or in a dependency array.

As you only need isKeyboard for item's styles you can probably replace `&.active.keyboard` with `body[data-awsui-focus-visible='true'] &.active`.

The `body[data-awsui-focus-visible='true']` is added when since the last keyboard event received globally there were no mouse clicks.
```

Reply:
Addressed in [48d4c6f](https://github.com/cloudscape-design/components/pull/804/commits/48d4c6f4c7200d01acf1e5390604762f82fbc881).